src/Interfaces.ts

```diff
@@ -17,45 +26,76 @@ interface ClaimContext {
   }
 }
 
+const signatureSchema = {
+  '@graph': Joi.object().keys({
+    '@type': Joi.string().only(Object.keys(jsigs.suites).map(suite => `https://w3id.org/security#${suite}`)),
+    created: Joi.object().keys({
+      '@type': Joi.string().uri(),
+      '@value': Joi.string()
+        .required()
+        .isoDate(),
+    }),
+    'http://purl.org/dc/terms/creator': Joi.object({
+      '@id': Joi.string()
+        .required()
+        .uri(),
+    }).required(),
+    'https://w3id.org/security#jws': Joi.string(),
+  }),
+}
+
+const claimSchema = Joi.object({
+  id: Joi.string()
+    .required()
+    .alphanum()
+    .min(64),
+  issuer: Joi.string()
+    .required()
+    .uri({
+      scheme: ['po.et', 'http', 'https', 'did', 'data'],
+    }),
+  issuanceDate: Joi.string()
+    .required()
+    .isoDate(),
+  type: Joi.string()
+    .required()
+    .only([ClaimType.Identity, ClaimType.Work]),
+  claim: Joi.object().required(),
+  'https://w3id.org/security#proof': Joi.object(signatureSchema).required(),
+})
+
 export function isClaim(object: any): object is Claim {
-  // TODO: use joi or protobuf
-  return !!(object.id && object.publicKey && object.signature && object.type && object.attributes && object.created)
+  return Joi.validate(object, claimSchema).error === null
 }
 
 // WARNING: This MUST account for ALL of the attributes in a Claim, except for id, @context, and signature.
 // Otherwise those attributes without context will be left out of the canonized/signed claim.
 // Refer to https://www.w3.org/2018/jsonld-cg-reports/json-ld/#the-context
+// TODO: make sure we allow customers to attach their own context
 export const ClaimContext: ClaimContext = {
   '@context': {
-    attributes: 'http://schema.org/CreativeWork',
+    issuer: 'http://schema.org/string',
+    issuanceDate: 'http://purl.org/dc/terms/created',
+    type: 'http://schema.org/additionalType',
+    claim: 'http://schema.org/CreativeWork',
```

Comment:
Is this supposed to work for all claims, or only creative work claims?

Reply:
hmmmm only Creative work claims, I have not found a verifiable claims definition. I may have to fall back to item.

Reply:
Can we make our own?

Reply:
We can. The verifiable claims definition is here: https://web-payments.org/contexts/credentials-v1.jsonld

As you can see, the context is still using `issued` instead of the newer `issuanceDate`.

But even that leaves the `claim` object completely open. They discuss defining your own in https://w3c.github.io/vc-data-model/#extensibility. However, this is more detailed than I think we want. (They recommend publishing the jsonld context document online for reference)

Reply:
For now this link is too specific. Do we have a plan for how to move forward?

Reply:
Yes,

1. I have set the context for `claim` to be `https://schema.org/Thing`. Everything in schema.org inherits from Thing.
2. For work claims, I set the `claim` to be `https://schema.org/CreativeWork`

From there, I believe that we need to go through our other potential claim types and try to figure out the schema for them. Identity claims need some thought sooner rather than later.